src/UI/Seller/src/app/shared/containers/user-table/user-table.component.ts

```diff
@@ -0,0 +1,143 @@
+import { ModalService } from '@app-seller/shared/services/modal/modal.service';
+import { forkJoin } from 'rxjs';
+
+@Component({
+  selector: 'user-table',
+  templateUrl: './user-table.component.html',
+  styleUrls: ['./user-table.component.scss'],
+})
+export class UserTableComponent extends BaseBrowse implements OnInit {
+  users: ListUser;
+  selectedUser: User;
+  faTrash = faTrashAlt;
+  faCircle = faCircle;
+  faPlusCircle = faPlusCircle;
+  createModalID = 'CreateUserModal';
+  editModalID = 'EditUserModal';
+  // Default Columns.
+  @Input()
+  columns = [
+    'ID',
+    'Username',
+    'FirstName',
+    'LastName',
+    'Email',
+    'Active',
+    'Delete',
+  ];
+
+  // Only use this when assigning users to user groups.
+  @Input() userGroupID: string;
+
+  constructor(
+    private ocUserService: OcUserService,
+    private ocUserGroupService: OcUserGroupService,
+    private modalService: ModalService,
+    @Inject(applicationConfiguration) private appConfig: AppConfig
+  ) {
+    // The BaseBrowse super class contains functionality for search, sort, and pagination.
+    super();
+  }
+
+  ngOnInit() {
+    this.loadData();
+  }
+
+  openCreateModal() {
+    this.modalService.open(this.createModalID);
+  }
+
+  openEditModal(user: User) {
+    this.selectedUser = user;
+    this.modalService.open(this.editModalID);
+  }
+
+  // Overrides a method in BaseBrowse
+  // TODO - I think this observable stuff can be made cleaner with operators
+  loadData(): void {
+    // this.requestOptions is inherited from BaseBrowse
+    this.ocUserService
+      .List(this.appConfig.buyerID, this.requestOptions)
+      .subscribe((users) => {
+        //const users = users;
+        if (this.columns.indexOf('Assign') < 0 || !this.userGroupID) {
+          return (this.users = users);
+        }
+        const queue = users.Items.map((user) => {
+          return this.ocUserGroupService.ListUserAssignments(
+            this.appConfig.buyerID,
+            {
+              userGroupID: this.userGroupID,
+              userID: user.ID,
+            }
+          );
+        });
+        forkJoin(queue).subscribe((res: any) => {
+          res = res.filter((group) => group.Items.length > 0);
+          res.forEach((group) => {
+            const index = users.Items.findIndex(
+              (user) => user.ID === group.Items[0].UserID
+            );
+            (users.Items[index] as any).Assigned = true;
+          });
+          this.users = users;
+        });
+      });
+  }
+
+  assignUser(userID: string, assigned: boolean) {
+    if (assigned) {
+      this.ocUserGroupService
+        .SaveUserAssignment(this.appConfig.buyerID, {
+          UserID: userID,
+          UserGroupID: this.userGroupID,
+        })
+        .subscribe();
+    } else {
+      this.ocUserGroupService
+        .DeleteUserAssignment(this.appConfig.buyerID, this.userGroupID, userID)
+        .subscribe();
```

Comment:
Could we be using the OnDestroy  pattern that we've been using here, since there is nothing that's unsubscribing the observable.

Reply:
Good to be thinking of that but it doesn't apply in this case. Because this Observable is an http request, when the request comes back, the observable.complete() method will be called and all subscribers will unsubscribe.   

https://stackoverflow.com/questions/35042929/is-it-necessary-to-unsubscribe-from-observables-created-by-http-methods

This is something Crhistian, Miranda and I went back and forth on before learning.